dask_ml/_partial.py

```diff
@@ -171,12 +172,15 @@ def fit(model, x, y, compute=True, **kwargs):
         x = x.rechunk(chunks=(x.chunks[0], sum(x.chunks[1])))
 
     nblocks = len(x.chunks[0])
+    order = list(range(nblocks))
+    random.shuffle(order)
```

Comment:
For reproducibility, Incremental should take a `random_state` and then use that for shuffling the data.

I also think the shuffling should be optionally enabled / disabled (not sure what the default should be) by a hyper parameter.

Finally, I'm not sure what the parameter name should be. Typically scikit-learn uses `shuffle=True/False`. We can either re-use that, or call it `shuffle_blocks`, and reserve `shuffle` for when we actually shuffle data between blocks.

Reply:
I think shuffling should be enabled by default. That's what sklearn default too, and it's what SGD theory points too.

I've used `shuffle_blocks`; that describes exactly what it does. We can use `shuffle` for full shuffling, and `shuffle_each_block` to shuffle in each block.